hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDMetadataWriteClient.java

```diff
@@ -63,6 +75,41 @@ public String createNewInstantTime() {
     return TimelineUtils.generateInstantTime(false, timeGenerator);
   }
 
+  @Override
+  public boolean commit(String instantTime, JavaRDD<WriteStatus> writeStatuses, Option<Map<String, String>> extraMetadata,
+                        String commitActionType, Map<String, List<String>> partitionToReplacedFileIds,
+                        Option<BiConsumer<HoodieTableMetaClient, HoodieCommitMetadata>> extraPreCommitFunc,
+                        Option<WriteStatusHandlerCallback> writeStatusHandlerCallbackOpt) {
+    context.setJobStatus(this.getClass().getSimpleName(), "Committing stats: " + config.getTableName());
+    // Triggering the dag for writes to metadata table.
+    // When streaming writes are enabled, writes to metadata may not call this method as the caller tightly controls the dag de-referencing.
+    // Even then, to initialize a new partition in Metadata table and for non incremental operations like insert_overwrite, etc, writes to metadata table
+    // will invoke this commit method.
+    List<WriteStatus> writeStatusesList = writeStatuses.map(writeStatus -> writeStatus.removeMetadataIndexStatsAndErrorRecordsTracking()).collect();
+    // Compute stats for the writes and invoke callback
+    AtomicLong totalRecords = new AtomicLong(0);
+    AtomicLong totalErrorRecords = new AtomicLong(0);
```

Comment:
nitpick: since we're not updating these values from multiple threads, the atomicLong seems like it is not required.

Reply:
These are used in lambda expressions, therefore had to use atomic references here.